lib/utils/compose.ts

```diff
@@ -94,22 +95,60 @@ export function createProject(
 	};
 }
 
+const getRequestRetryParameters = (): RetryParametersObj => {
+	const testIntVar = (varName: string) => {
+		if (
+			!process.env.BALENA_CLI_TEST_TYPE ||
+			!varName.startsWith('BALENARCTEST_')
+		) {
+			// We only read the test env vars when in test mode.
+			return;
+		}
+		const { intVar } =
+			require('@balena/env-parsing') as typeof import('@balena/env-parsing');
+		return intVar(varName);
+	};
+	// We use the BALENARCTEST namespace and only parse the env vars while in test mode
+	// since we plan to switch all pinejs clients with the one of the SDK and might not
+	// BALENARCTEST to have to support these env vars.
+	return {
+		minDelayMs: testIntVar('BALENARCTEST_API_RETRY_MIN_DELAY_MS') ?? 1000,
+		maxDelayMs: testIntVar('BALENARCTEST_API_RETRY_MAX_DELAY_MS') ?? 60000,
+		maxAttempts: testIntVar('BALENARCTEST_API_RETRY_MAX_ATTEMPTS') ?? 7,
```

Comment:
Since on CI the tests run the cli on a separate sub-process, we can't mock these values, so the easiest way to workaround that was to add & start parsing a "tests only" env var (which we can later choose to surface to users if we want, but check the comment on why I think we shouldn't).